exporter/elasticsearchexporter/integrationtest/datareceiver.go

```diff
@@ -217,9 +225,21 @@ func (es *mockESReceiver) Start(ctx context.Context, host component.Host) error
 	r.HandleFunc("/_bulk", func(w http.ResponseWriter, r *http.Request) {
 		if !es.config.DecodeBulkRequests {
 			fmt.Fprintln(w, "{}")
+			defer r.Body.Close()
+			s := bufio.NewScanner(r.Body)
+			var cnt int64
+			for s.Scan() {
+				cnt++
+			}
+			if es.config.DocCount != nil {
+				es.config.DocCount.Add(cnt / 2) // 1 line for action, 1 line for document
+			}
```

Comment:
```suggestion
			defer r.Body.Close()
			// TODO: This code only handles uncompressed requests
			for s := bufio.NewScanner(r.Body); s.Scan(); {
				action := gjson.Get(s.Text(), "index._index")
				if !action.Exists() {
					// might be the last newline, skip
					continue
				}
				switch action.Str {
				case TestLogsIndex:
					es.logsConsumer.ConsumeLogs(context.Background(), emptyLogs)
				case TestTracesIndex:
					es.tracesConsumer.ConsumeTraces(context.Background(), emptyTrace)
				}
				// skip next line
				s.Scan()
			}
			fmt.Fprintln(w, "{}")
			return
```
Since we are scanning anyway we can do better and utilize the next consumer in the chain (in addition to removing the `DocCount` as in another comment).